wordpress-mobile/WordPress-Android
# Posts: Enable author filter

Parent #19339 

This PR hooks up the Author filter for posts list:
Everyone Selected | Me Selected
-- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/a97c9ead-868b-4746-a2c2-32eae4fe5a23"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/221fb2a1-569c-4897-8726-37a9cd35e5b7">

Everyone Selected | Me Selected
-- | --
<img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/5a5519db-c17a-4676-a769-124aa00001ba"> | <img width="250" height="500" alt="Alt desc" src="https://github.com/wordpress-mobile/WordPress-Android/assets/506707/15421207-249d-4aae-92e2-62053494bedb">

**Note**: 
The additional check for a site having more than 1 user with come in a later PR because it requires a FluxC change.
This PR may break Pages because they used shared layout files
The people icon is not showing in light mode (this will be in a separate PR)
 
**To test:**
- Do a fresh install
- Login 
- Select a site in which you have edit capabilities
- Navigate to My Site > Posts
- ✅ Verify the author filter icon shows the "everyone" icon 
- Tap on the author filter
- ✅ Verify the author filter is opened and "everyone" is selected
- Change the selection to Me
- ✅ Verify the author filter icon shows the "me" avatar (if there is an avatar)
- Navigate back to My Site
- Navigate back into Posts
- ✅ Verify the author filter icon shows the last selection you made (which if you followed these steps, should be "me"
- Swipe the app closed
- Reopen the app 
- Navigate to My Site > Posts
- ✅ Verify the author filter icon shows the "me" avatar

## Regression Notes
1. Potential unintended areas of impact
The author filter doesn't work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and `PostListMainViewModel`

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

UI Changes testing checklist:
